### ISSUE TITLE
Bug: 'Remove this file' does not work properly

### ISSUE THREAD

**Issue by Lionqueen94** [surveyjs/survey-library#5070]
### Are you requesting a feature, reporting a bug or asking a question?
Reporting a bug

### What is the current behavior?
In version 1.9.54:
When uploading a file with an invalid filename (in our case filenames with a comma for example), you cannot remove the file anymore.

### What is the expected behavior?
In version 1.9.53:
When uploading a file with an invalid filename (in our case filenames with a comma for example), you can remove the file.

### How would you reproduce the current behavior (if this is a bug)?
See this Plunker:
https://plnkr.co/edit/s8yttsPa6pqRw4VQ
Switch between 1.9.53 and 1.9.54 and see the behavior when uploading a file and then trying to remove it. In 1.9.54 the file does not remove.
In the case of the Plunker all files do not remove it seems, in my production case files with valid filenames do remove, but with invalid filenames don't. Filename validation is done in the backend btw, so I cannot add that code and also that does not seem to be the problem here.

#### Provide the test code and the tested page URL (if applicable)
https://plnkr.co/edit/s8yttsPa6pqRw4VQ

### Specify your
* browser: chrome/firefox
* browser version: latest
* surveyjs platform (angular or react or jquery or knockout or vue): jquery
* surveyjs version: 1.9.54

**Comment by tsv2013** [comment]
You need to catch the "error" case (invalid filename) during file upload
Instead of
```JS
xhr.onload = function () {
    var data = JSON.parse(xhr.responseText);
    options.callback("success", options.files.map(function (file) {
      return {
        file: file,
        content: data[file.name]
      };
    }));
  };
```

you need to use something like
```JS
xhr.onload = function () {
    var data = JSON.parse(xhr.responseText);
    options.callback("success", options.files.map(function (file) {
      return {
        file: file,
        content: data[file.name]
      };
    }));
  };
xhr.onerror = function () {
    options.callback("error");
};
```

In case of this deosn't help please share the exact steps to reproduce the issue with your example. Unfortunately right now it unclear how to reproduce the issue.

**Comment by Lionqueen94** [comment]
@tsv2013 Please check the Plunker. Upload a file and click on the 'remove this file' button. In v1.9.53 the file removes, in v1.9.54 it doesn't. Clicking on Clear does remove the file in both cases. I'm pretty sure something went wrong in the last release, since there were changes made in the file question code.

**Comment by tsv2013** [comment]
@Lionqueen94 What should be the file name in order to reproduce the issue? Can you share a sample one?

**Comment by Lionqueen94** [comment]
@tsv2013 See here what I mean in the Plunker (https://plnkr.co/edit/s8yttsPa6pqRw4VQ)
In v1.9.53 removing a file works as expected with both the remove as well as the Clean button:

https://user-images.githubusercontent.com/2974080/196555682-be7ae520-14e1-4509-8fbd-6b45075f353f.mp4

However in v1.9.54 removing a file does not work, it only works via the Clear button:

https://user-images.githubusercontent.com/2974080/196555687-d1a75277-20c5-43a2-b84c-1120433440c0.mp4

**Comment by Lionqueen94** [comment]
@tsv2013 Any idea why the file removal is acting weird in the latest version? See the screen-captures in my previous comment.

**Comment by Lionqueen94** [comment]
Seems to be fixed in https://github.com/surveyjs/survey-library/issues/5115 I'll test this once it's released.

**Comment by Lionqueen94** [comment]
It is indeed fixed in 1.9.57, closing this issue.
